src/charm.py

```diff
@@ -746,7 +821,7 @@ def _reconcile_mongo_hosts_and_users(self, event: RelationEvent) -> None:
                 event.defer()
 
     def _on_stop(self, event) -> None:
-        if "True" == self.unit_peer_data.get("unit_departed", "False"):
+        if self.unit_departed:
```

Comment:
what is the significance of this / where else is it used?

Reply:
I just refactored the `unit_departed` usage.
It's used to leave time for the unit to step down from the replica set.
It's set to `True` in the `relation_changed` handler in the case we're in a `RelationDepartedEvent`.